include/highfive/h5easy_bits/H5Easy_scalar.hpp

```diff
@@ -16,27 +16,34 @@ namespace H5Easy {
 
 namespace detail {
 
-namespace scalar {
+/*
+This base class presents fallback methods which will be automatically
+called from specializations, if not defined in the specialization.
+Every specialization of io_impl<T> should inherit from io_impl_base<T>.
+
+The full API set is:
+
+* dump
+* overwrite
+* dump_extend
+* load
+* load_part
 
-// create DataSet and write data
-template <class T, class E = void>
-struct dump_impl
+*/
+template<typename T>
+struct io_impl_base
```

Comment:
What is the reason to this extra name? Is seems that it requires you to make an alias below. I would simply use one name